From: Ben Hutchings <ben@decadent.org.uk>
Date: Thu, 22 Aug 2019 19:26:31 +0100
Subject: powerpc: Avoid ABI change for disabling TM
Forwarded: not-needed

Keep the TM state in struct paca_struct and struct thread_struct even
though we won't be using it any more.

The KVM state structures will change, but we ignore ABI changes there
since they won't affect out-of-tree modules.

---
Index: linux/arch/powerpc/include/asm/paca.h
===================================================================
--- linux.orig/arch/powerpc/include/asm/paca.h
+++ linux/arch/powerpc/include/asm/paca.h
@@ -173,7 +173,7 @@ struct paca_struct {
 	u8 pmcregs_in_use;		/* pseries puts this in lppaca */
 #endif
 	u64 sprg_vdso;			/* Saved user-visible sprg */
-#ifdef CONFIG_PPC_TRANSACTIONAL_MEM
+#if 1 /* ifdef CONFIG_PPC_TRANSACTIONAL_MEM */
 	u64 tm_scratch;                 /* TM scratch area for reclaim */
 #endif
 
Index: linux/arch/powerpc/include/asm/processor.h
===================================================================
--- linux.orig/arch/powerpc/include/asm/processor.h
+++ linux/arch/powerpc/include/asm/processor.h
@@ -293,7 +293,7 @@ struct thread_struct {
 					   call or trap return */
 	int		used_spe;	/* set if process has used spe */
 #endif /* CONFIG_SPE */
-#ifdef CONFIG_PPC_TRANSACTIONAL_MEM
+#if 1 /* ifdef CONFIG_PPC_TRANSACTIONAL_MEM */
 	u8	load_tm;
 	u64		tm_tfhar;	/* Transaction fail handler addr */
 	u64		tm_texasr;	/* Transaction exception & summary */
